dbt_platform_helper/providers/schema_migrations.py

```diff
@@ -0,0 +1,106 @@
+from collections import Counter
+from copy import deepcopy
+from typing import Protocol
+
+from dbt_platform_helper.platform_exception import PlatformException
+
+
+class InvalidMigrationConfigurationException(PlatformException):
+    pass
+
+
+class PlatformConfigSchemaMigrationException(PlatformException):
+    pass
+
+
+class SchemaMigrationProtocol(Protocol):
+    def from_version(self) -> int: ...
+
+    def migrate(self, platform_config: dict) -> dict: ...
+
+    def messages(self) -> list[str]: ...
+
+
+class SchemaV0ToV1Migration:
```

Comment:
I would be inclined to put the actual migration classes in individual files in a subdirectory.